src/middle/Stan_math_signatures.ml

```diff
@@ -95,9 +95,25 @@ let reduce_sum_slice_types =
   List.concat (List.map ~f:base_slice_type reduce_sum_allowed_dimensionalities)
 
 (* Variadic ODE *)
+let variadic_ode_adjoint_ctl_tol_arg_types =
+  [ (UnsizedType.DataOnly, UnsizedType.UReal)
+    (* real relative_tolerance_forward *)
+  ; (DataOnly, UVector) (* vector absolute_tolerance_forward *)
+  ; (DataOnly, UReal) (* real relative_tolerance_backward *)
+  ; (DataOnly, UVector) (* real absolute_tolerance_backward *)
+  ; (DataOnly, UReal) (* real relative_tolerance_quadrature *)
+  ; (DataOnly, UReal) (* real absolute_tolerance_quadrature *)
+  ; (DataOnly, UInt) (* int max_num_steps *)
+  ; (DataOnly, UInt) (* int num_steps_between_checkpoints *)
+  ; (DataOnly, UInt) (* int interpolation_polynomial *)
+  ; (DataOnly, UInt) (* int solver_forward *)
+  ; (DataOnly, UInt)
+  (* int solver_backward *)
+   ]
+
 let variadic_ode_tol_arg_types =
-  [ (UnsizedType.AutoDiffable, UnsizedType.UReal)
-  ; (AutoDiffable, UReal); (DataOnly, UInt) ]
+  [ (UnsizedType.DataOnly, UnsizedType.UReal)
+  ; (DataOnly, UReal); (DataOnly, UInt) ]
```

Comment:
These are for nonadjoint ODEs. Why did they change?

Reply:
This is a bug currently. I should have mention it in the PR comment, sorry. 

The tolerances in the nonadjoint should not allow autodiffable arguments as that causes C++ not to compile. This was reported by @bbbales2 in the Math PR but that PR has 500+ comments and I cant find the exact comment right now.

This would now error in stanc3 which is better I guess.